### PR TITLE
[test gap] add a new test case for bgp session check

### PR DESCRIPTION
**Pull request by lipxu**
### Description of PR

Summary:
Fixes # (issue)
26557087

### Type of change

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] Test case(new/improvement)

### Back port request
- [ ] 202012
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405

#### What is the motivation for this PR?
FRR issue https://github.com/sonic-net/sonic-buildimage/issues/12380
Verify the bgp sessions' status

#### How did you do it?
    1: check all bgp sessions are up
    2: inject failure, shutdown fanout physical interface or neighbor port
    4: do the test, reset bgp or swss or do the reboot
    5: Verify all bgp sessions are up

#### How did you verify/test it?
Run the test case

#### Supported testbed topology if it's a new test case?
T0,T1

### CODE REVIEW

**Review by yutongzhang-microsoft — CHANGES_REQUESTED** [review]
Hi, @lipxu, have you tested this script on kvm testbed?

**Comment by lipxu** [comment]
> Hi, @lipxu, have you tested this script on kvm testbed? If this script can run on kvm, please add it into corresponding PR checker.

Not yet, but the script includes a warm-reboot operation, not sure whether it is supported on kvm.
Could you suggest a quick way to verify it? thanks a lot

File: tests/bgp/test_bgp_session.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+    elif failure_type == "neighbor":
+        logger.info("###### shutdown neighbor {}".format(failure_type))
+        for port in local_interfaces:
+            neighbor_port = setup['neighhosts'][neighbor]['interface'][port]['port']
+            logger.info("shutdown interface neighbor {} port {}".format(neighbor_name, neighbor_port))
+            nbrhosts[neighbor_name]['host'].shutdown(neighbor_port)
+            time.sleep(1)
+
+    duthost.shell('show ip bgp summary', module_ignore_errors=True)
```
Comment (lolyu) [inline tests/bgp/test_bgp_session.py:126]: do we need to verify the bgp sessions are down here?

**Comment by yutongzhang-microsoft** [comment]
> ncludes a warm-reboot operation, not sure whethe

Hi, @lipxu , can you test locally first? And then, you can add this script into onboarding T0 PR checker https://github.com/sonic-net/sonic-mgmt/blob/master/.azure-pipelines/pr_test_scripts.yaml#L343

File: tests/bgp/test_bgp_session.py (hunk 1 above)
Comment (lipxu) [reply tests/bgp/test_bgp_session.py:134]: Thanks @lolyu , will add a check here.

**Comment by lipxu** [comment]
> > ncludes a warm-reboot operation, not sure whethe
> 
> Hi, @lipxu , can you test locally first? And then, you can add this script into onboarding T0 and T1 PR checker https://github.com/sonic-net/sonic-mgmt/blob/master/.azure-pipelines/pr_test_scripts.yaml#L343

This case includes fanout interface operation (shutdown/no shutdown), and warm-reboot which not supported on kvm yet. will skip it on kvm.

**Comment by lipxu** [comment]
> > ncludes a warm-reboot operation, not sure whethe
> 
> Hi, @lipxu , can you test locally first? And then, you can add this script into onboarding T0 and T1 PR checker https://github.com/sonic-net/sonic-mgmt/blob/master/.azure-pipelines/pr_test_scripts.yaml#L343

Hi, @yutongzhang-microsoft added the test case into the PR test list.

**Review by yutongzhang-microsoft — APPROVED** [review]

**Review by StormLiangMS — APPROVED** [review]
LGTM

**Pull request merged**